test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1304 @@
+	if ver, verr := semver.ParseTolerant(match[1]); verr == nil {
+		return ver.Compare(semver.MustParse(MinContainerRuntimeVersion)) >= 0
+	}
+	return false
+}
+
+func getTestResourceInfo(tcInfo TestContainerInfo) (v1.ResourceRequirements, v1.ResourceList, []v1.ContainerResizePolicy) {
+	var res v1.ResourceRequirements
+	var alloc v1.ResourceList
+	var resizePol []v1.ContainerResizePolicy
+
+	if tcInfo.Resources != nil {
+		var lim, req v1.ResourceList
+		if tcInfo.Resources.CPULim != "" || tcInfo.Resources.MemLim != "" || tcInfo.Resources.EphStorLim != "" {
+			lim = make(v1.ResourceList)
+		}
+		if tcInfo.Resources.CPUReq != "" || tcInfo.Resources.MemReq != "" || tcInfo.Resources.EphStorReq != "" {
+			req = make(v1.ResourceList)
+		}
+		if tcInfo.Resources.CPULim != "" {
+			lim[v1.ResourceCPU] = resource.MustParse(tcInfo.Resources.CPULim)
+		}
+		if tcInfo.Resources.MemLim != "" {
+			lim[v1.ResourceMemory] = resource.MustParse(tcInfo.Resources.MemLim)
+		}
+		if tcInfo.Resources.EphStorLim != "" {
+			lim[v1.ResourceEphemeralStorage] = resource.MustParse(tcInfo.Resources.EphStorLim)
+		}
+		if tcInfo.Resources.CPUReq != "" {
+			req[v1.ResourceCPU] = resource.MustParse(tcInfo.Resources.CPUReq)
+		}
+		if tcInfo.Resources.MemReq != "" {
+			req[v1.ResourceMemory] = resource.MustParse(tcInfo.Resources.MemReq)
+		}
+		if tcInfo.Resources.EphStorReq != "" {
+			req[v1.ResourceEphemeralStorage] = resource.MustParse(tcInfo.Resources.EphStorReq)
+		}
+		res = v1.ResourceRequirements{Limits: lim, Requests: req}
+	}
+	if tcInfo.Allocations != nil {
+		alloc = make(v1.ResourceList)
+		if tcInfo.Allocations.CPUAlloc != "" {
+			alloc[v1.ResourceCPU] = resource.MustParse(tcInfo.Allocations.CPUAlloc)
+		}
+		if tcInfo.Allocations.MemAlloc != "" {
+			alloc[v1.ResourceMemory] = resource.MustParse(tcInfo.Allocations.MemAlloc)
+		}
+		if tcInfo.Allocations.ephStorAlloc != "" {
+			alloc[v1.ResourceEphemeralStorage] = resource.MustParse(tcInfo.Allocations.ephStorAlloc)
+		}
+
+	}
+	if tcInfo.CPUPolicy != nil {
+		cpuPol := v1.ContainerResizePolicy{ResourceName: v1.ResourceCPU, RestartPolicy: *tcInfo.CPUPolicy}
+		resizePol = append(resizePol, cpuPol)
+	}
+	if tcInfo.MemPolicy != nil {
+		memPol := v1.ContainerResizePolicy{ResourceName: v1.ResourceMemory, RestartPolicy: *tcInfo.MemPolicy}
+		resizePol = append(resizePol, memPol)
+	}
+	return res, alloc, resizePol
+}
+
+func initDefaultResizePolicy(containers []TestContainerInfo) {
+	noRestart := v1.NotRequired
+	setDefaultPolicy := func(ci *TestContainerInfo) {
+		if ci.CPUPolicy == nil {
+			ci.CPUPolicy = &noRestart
+		}
+		if ci.MemPolicy == nil {
+			ci.MemPolicy = &noRestart
+		}
+	}
+	for i := range containers {
+		setDefaultPolicy(&containers[i])
+	}
+}
+
+func makeTestContainer(tcInfo TestContainerInfo) (v1.Container, v1.ContainerStatus) {
+	cmd := "grep Cpus_allowed_list /proc/self/status | cut -f2 && sleep 1d"
```

Comment:
It might be better to add `trap exit TERM;` to this command. Otherwise, I guess it would take 30 seconds to restart a container in test cases where `ResizePolicy` is `RestartContainer`.

Reply:
Thanks, good point, acknowledged will commit, test and push update.

Reply:
Hi, adding trap exit TERM, results in failure later when attempting to verify group values. I think we need to keep it as is and live with the 30 seconds delay.

Reply:
the command doesn't need to be a one liner, can be a more complex script; there should be examples in the codebase.
Depending on how often we resize, I do think accepting a timeout initially is not perfect but still acceptable